zephyr/include/sof/auth/auth_api.h

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+ *         Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+ */
+#ifndef AUTH_API_H
+#define AUTH_API_H
+
+#include "auth_api_iface.h"
+
+#define AUTH_API_CALLBACKS_ADDR (0x162000 + 0x140)
+
+/* The same as auth_api->init. */
+inline int auth_api_init(struct auth_api_ctx *ctx,
+			 void *scratch_buff,
+			 size_t scratch_buff_size,
+			 enum auth_image_type image_type)
+{
+	struct auth_api_ctx **auth_api_callbacks =
+			(struct auth_api_ctx **)(AUTH_API_CALLBACKS_ADDR);
+
+	ctx->version_api = (*auth_api_callbacks)->version_api;
+	ctx->auth_api = (*auth_api_callbacks)->auth_api;
+	return ctx->auth_api->init(ctx, scratch_buff, scratch_buff_size, image_type);
+}
+
+/* The same as auth_api->cleanup. */
```

Comment:
I guess all the ops are mandatory ? i.e. any auth driver need to have them all implemented.

Reply:
This is API provided by ROM_EXT.